### PR TITLE
`sage.calculus.expr`: Split out from `sage.calculus.all`

### PR DESCRIPTION
**Pull request by mkoeppe**
Cherry-picked from #36676/#37900.

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by mkoeppe** [comment]
Split out from https://github.com/sagemath/sage/pull/36676 / https://github.com/sagemath/sage/pull/37900, previously reviewed 6 months ago.

File: src/sage/calculus/expr.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        Traceback (most recent call last):
+        ...
+        TypeError: unable to convert <function function_with_keyword_only_arg at 0x...>
+        to a symbolic expression
+    """
+    if isinstance(x, Expression):
+        return x
+    elif hasattr(x, '_symbolic_'):
+        return x._symbolic_(SR)
+    elif isinstance(x, (tuple, list, FreeModuleElement)):
```
Comment (jhpalmieri) [inline src/sage/calculus/expr.py:173]: I'm curious, why move away from using `is_FreeModuleElement` but keep using `is_Matrix` (line 181)?

Comment (jhpalmieri) [reply src/sage/calculus/expr.py:173]: (Otherwise this looks fine to me.)

Comment (mkoeppe) [reply src/sage/calculus/expr.py:173]: Could as well also replace `is_Matrix`, but that is defined (along with the ABC `Matrix`) in the depths of `sage.structure.element`, which is shipped already with the lowest-level distribution, **sagemath-objects**, so it is not relevant for the modularization.

Comment (mkoeppe) [reply src/sage/calculus/expr.py:173]: ... also note that in the systematic deprecation of `is_...` functions, I have taken care of parent classes primarily so far.

Comment (jhpalmieri) [reply src/sage/calculus/expr.py:173]: Thanks for the explanation, that makes sense.

**Comment by jhpalmieri** [comment]
Let's merge it.

**Comment by mkoeppe** [comment]
Thanks again.

**Pull request merged**